stdlib/Dates/test/arithmetic.jl

```diff
@@ -474,23 +474,23 @@ end
-        @test (Dates.Hour(1) + Dates.Minute(1)) + t3 == [Dates.DateTime(2009, 1, 1, 1, 1), Dates.DateTime(2009, 1, 2, 1, 1), Dates.DateTime(2009, 1, 3, 1, 1)]
-        @test t1 + (Dates.Month(1) + Dates.Day(1)) == [Dates.Date(2009, 2, 2) Dates.Date(2009, 2, 3) Dates.Date(2009, 2, 4); Dates.Date(2009, 3, 2) Dates.Date(2009, 3, 3) Dates.Date(2009, 3, 4)]
-        @test t3 + (Dates.Hour(1) + Dates.Minute(1)) == [Dates.DateTime(2009, 1, 1, 1, 1), Dates.DateTime(2009, 1, 2, 1, 1), Dates.DateTime(2009, 1, 3, 1, 1)]
-        @test t1 - Dates.Day(1) == [Dates.Date(2008, 12, 31) Dates.Date(2009, 1, 1) Dates.Date(2009, 1, 2); Dates.Date(2009, 1, 31) Dates.Date(2009, 2, 1) Dates.Date(2009, 2, 2)]
-        @test t3 - Dates.Hour(1) == [Dates.DateTime(2008, 12, 31, 23), Dates.DateTime(2009, 1, 1, 23), Dates.DateTime(2009, 1, 2, 23)]
-        @test t1 - (Dates.Month(1) + Dates.Day(1)) == [Dates.Date(2008, 11, 30) Dates.Date(2008, 12, 1) Dates.Date(2008, 12, 2); Dates.Date(2008, 12, 31) Dates.Date(2009, 1, 1) Dates.Date(2009, 1, 2)]
-        @test t3 - (Dates.Hour(1) + Dates.Minute(1)) == [Dates.DateTime(2008, 12, 31, 22, 59), Dates.DateTime(2009, 1, 1, 22, 59), Dates.DateTime(2009, 1, 2, 22, 59)]
+        @test_deprecated "use `x .- y` instead" Dates.Date(2010, 1, 1) - t1
+        @test_deprecated "use `x .- y` instead" t1 - Dates.Date(2010, 1, 1)
+        @test_deprecated "use `x .- y` instead" Dates.DateTime(2009, 1, 1) - t3
+        @test_deprecated "use `x .- y` instead" t3 - Dates.DateTime(2009, 1, 1)
+        @test_deprecated "use `x .+ y` instead" Dates.Day(1) + t1
+        @test_deprecated "use `x .+ y` instead" Dates.Hour(1) + t3
+        @test_deprecated "use `x .+ y` instead" t1 + Dates.Day(1)
+        @test_deprecated "use `x .+ y` instead" t3 + Dates.Hour(1)
+        @test_deprecated "use `x .+ y` instead" t5 + Dates.Hour(1)
+        @test_deprecated "use `x .+ y` instead" (Dates.Month(1) + Dates.Day(1)) + t1
+        @test_deprecated "use `x .+ y` instead" (Dates.Hour(1) + Dates.Minute(1)) + t3
+        @test_deprecated "use `x .+ y` instead" t1 + (Dates.Month(1) + Dates.Day(1))
+        @test_deprecated "use `x .+ y` instead" t3 + (Dates.Hour(1) + Dates.Minute(1))
+        @test_deprecated "use `x .- y` instead" t1 - Dates.Day(1)
+        @test_deprecated "use `x .- y` instead" t3 - Dates.Hour(1)
+        @test_deprecated "use `x .- y` instead" t1 - (Dates.Month(1) + Dates.Day(1))
+        @test_deprecated "use `x .- y` instead" t3 - (Dates.Hour(1) + Dates.Minute(1))
         @test t2 - t1 == [Dates.Day(1) Dates.Day(31) Dates.Day(365); Dates.Day(365) Dates.Day(28) Dates.Day(1)]
         @test t4 - t3 == [Dates.Millisecond(1000), Dates.Millisecond(60000), Dates.Millisecond(3600000)]
         @test (Dates.Date(2009, 1, 1):Dates.Week(1):Dates.Date(2009, 1, 21)) - (Dates.Date(2009, 1, 1):Dates.Day(1):Dates.Date(2009, 1, 3)) == [Dates.Day(0), Dates.Day(6), Dates.Day(12)]
```

Comment:
```jl
julia> typeof((Dates.DateTime(2009, 1, 1, 1, 1, 1):Dates.Second(1):Dates.DateTime(2009, 1, 1, 1, 1, 3)))
StepRange{DateTime, Second}

julia> methods(-, (StepRange{DateTime, Second},StepRange{DateTime, Second}))
# 1 method for generic function "-" from Base:
 [1] -(x::OrdinalRange{T}, y::OrdinalRange{T}) where T<:TimeType
     @ Dates /mnt/v/julia/usr/share/julia/stdlib/v1.10/Dates/src/arithmetic.jl:95
```
https://github.com/JuliaLang/julia/blob/78fbf1bd74ded2a8a830f9f4b35c98666f2a7e16/stdlib/Dates/src/arithmetic.jl#L94-L96

Perhaps this usage is not deprecated.